src/vivarium_gates_mncnh/data/hemoglobin_effects/hgb_lbwsg_effect_generation.py

```diff
@@ -0,0 +1,371 @@
+                        rr_data=ptb_rrs,
+                        location=location,
+                        outcome="ptb",
+                        sex=sex,
+                        draw=draw,
+                        exposure_level=exposure_level,
+                    )
+                )
+                bw_shifts.append(
+                    calculate_shift_from_rr(
+                        exp_data=exp,
+                        rr_data=lbw_rrs,
+                        location=location,
+                        outcome="lbw",
+                        sex=sex,
+                        draw=draw,
+                        exposure_level=exposure_level,
+                    )
+                )
+            temp_ga = pd.DataFrame(
+                {
+                    "location": location,
+                    "sex": sex,
+                    "draw": draw,
+                    "exposure": exposure_levels,
+                    "value": ga_shifts,
+                    "outcome": "gestational_age",
+                }
+            )
+            temp_bw = pd.DataFrame(
+                {
+                    "location": location,
+                    "sex": sex,
+                    "draw": draw,
+                    "exposure": exposure_levels,
+                    "value": bw_shifts,
+                    "outcome": "birth_weight",
+                }
+            )
+            results = pd.concat([results, temp_ga, temp_bw], ignore_index=True)
+    return results
+
+
+def scale_lbwsg_to_iv_iron(lbwsg_shifts, draw):
+    """Scales hemoglobin effects on GA and BW (relative to the hemoglobin TMREL) to the effect size of IV iron"""
+
+    # TODO: read in IV iron effect size/threshold from the repo/artifact here instead of hard-coding it
+    iv_iron_md = 23  # change in hemoglobin exposure associated with IV iron
+    iv_iron_threshold = 100  # maximum hemoglobin exposure level (g/L) eligible for IV iron
+
+    lbwsg_shifts["exposure"] = lbwsg_shifts["exposure"]
+    lbwsg_shifts = lbwsg_shifts.sort_values(
+        by=[x for x in lbwsg_shifts.columns if x not in ["exposure", "value"]] + ["exposure"]
+    )
+
+    exposure_levels = sorted(lbwsg_shifts.exposure.unique().tolist())
+    assert np.all(
+        exposure_levels[i] < exposure_levels[i + 1] for i in range(len(exposure_levels) - 1)
+    ), "exposure levels not sorted in ascending order"
+    for location in lbwsg_shifts.location.unique():
+        for sex in lbwsg_shifts.sex.unique():
+            for outcome in lbwsg_shifts.outcome.unique():
+                assert np.all(
+                    lbwsg_shifts.loc[
+                        (lbwsg_shifts.location == location)
+                        & (lbwsg_shifts.sex == sex)
+                        & (lbwsg_shifts.outcome == outcome)
+                    ].exposure.values
+                    == exposure_levels
+                ), "Inconsistencies in exposure level values and/or orders"
+    assert np.allclose(
```

Comment:
I would not have an assert in code you are executing at runtime. If you are worried about something happening then you should throw an error since an error and a failed assertion will both kill the program.

Reply:
There is [plenty of debate about this](https://discuss.python.org/t/stop-ignoring-asserts-when-running-in-optimized-mode/13132/13) but an `assert` in Python can be seen as just syntactic sugar for raising an exception. I feel like making these errors would just make the code longer and harder to read.

Reply:
Ok, keeping as is!